internal/commands/app/create.go

```diff
@@ -199,9 +230,9 @@ func (cmd *CommandCreate) Handler(profile *user.Profile, ui terminal.UI, clients
 	}
 
 	if cmd.inputs.Template == "" {
-		return cmd.handleCreateApp(profile, ui, clients, groupID, rootDir, appRemote, dsClusters, dsDatalakes)
+		return cmd.handleCreateApp(profile, ui, clients, groupID, rootDir, appRemote, dsClusters, dsDatalakes, dsServerlessInstances)
 	}
-	return cmd.handleCreateTemplateApp(profile, ui, clients, groupID, rootDir, dsClusters, dsDatalakes)
+	return cmd.handleCreateTemplateApp(profile, ui, clients, groupID, rootDir, dsClusters, dsDatalakes, dsServerlessInstances)
```

Comment:
Yeah I thought so too - I only added serverless instances here to align with how the code currently looks for datalakes. I can remove both of those from `handleCreateTemplateApp` if that works for you

Reply:
Yeah I feel like we should remove them, and maybe also comment on the `handleCreateTemplateApp` function. That way the assumption that template apps are not compatible with serverless data sources will more explicit and clear. This could also prevent someone from adding additional unnecessary logic for serverless data sources in that function

Reply:
can we check in quickly with @andrewzlchen on this?

looks like we added clusters and datalakes to this function back in: https://github.com/10gen/realm-cli/pull/187/files

I just want to make sure we're not breaking anything existing (I wish I had more insight into the template app side of things)

Reply:
Yup, I think we can keep the `dsClusters` and `dsDatalakes` arguments.

I remember chatting with Melodee on why Serverless doesn't work for TSAs (but the details on why that is eludes me) so removing this arg makes sense to me.

While TSAs are not supporting Datalakes today, this isn't something that is set in stone. The main reason why it's not supported was that it was a decent amount of work that wasn't originally scoped in the project.

I'd like to come back around to supporting datalakes in TSAs at some point, but I haven't heard too much about customers requesting this yet.

Reply:
Thanks Andrew, that makes sense! I'll keep datalakes but remove serverless here.

If you were curious, Serverless instances currently don't support change streams, which all TSAs use in some way (either through sync/triggers, or by opening a change stream in the frontend code).